crates/runtime/src/dataconnector/databricks.rs

```diff
@@ -94,40 +115,140 @@ impl Databricks {
-                let databricks_spark = DatabricksSparkConnect::new(
-                    endpoint.to_string(),
-                    cluster_id.expose_secret().to_string(),
-                    token.expose_secret().to_string(),
+
+                let databricks_use_ssl = match params.get("use_ssl").expose().ok() {
+                    Some(value) => match value {
+                        "true" => true,
+                        "false" => false,
+                        _ => return InvalidUsesslSnafu { value }.fail(),
+                    },
+                    None => true, // Default value
+                };
+
+                Self::build_spark_connect_connector(
+                    endpoint,
+                    auth_credentials,
+                    token_provider_registry,
+                    cluster_id,
                     databricks_use_ssl,
                 )
                 .await
-                .context(UnableToConstructDatabricksSparkSnafu)?;
-                Ok(Self {
-                    read_provider: Arc::new(databricks_spark.clone()),
-                })
             }
             _ => Err(Error::InvalidMode {
                 value: mode.to_string(),
             }),
         }
     }
 
+    pub fn build_auth_credentials(params: &Parameters) -> Result<AuthCredentials> {
+        let token = params.get("token").ok();
+        let client_id = params.get("client_id").expose().ok();
+        let client_secret = params.get("client_secret").ok();
+
+        match (token, client_id, client_secret) {
+            (Some(token), None, None) => Ok(AuthCredentials::Token(token)),
+            (None, Some(client_id), Some(client_secret)) => {
+                Ok(AuthCredentials::ServicePrincipal(client_id, client_secret))
+            }
+            (None, None, None) => {
+                InvalidConfigurationSnafu {
+                    message: "Missing `databricks_token` or `databricks_client_id` and `databricks_client_secret` parameters".to_string(),
+                }
+                .fail()
+            }
+            (None, Some(_), None) => {
+                MissingParameterSnafu {
+                    parameter: "`databricks_client_secret`".to_string(),
+                }
+                .fail()
+            }
+            (None, None, Some(_)) => {
+                MissingParameterSnafu {
+                    parameter: "databricks_client_id".to_string(),
+                }
+                .fail()
+            }
+            (Some(_), Some(_), Some(_)) => {
+                InvalidConfigurationSnafu {
+                    message: "Choose either `databricks_token` or `databricks_client_id` and `databricks_client_secret`".to_string(),
+                }
+                .fail()
+            }
+            _ => {
+                InvalidConfigurationSnafu {
+                    message: format!("Invalid authentication configuration: {params:?}"),
```

Comment:
`params` contains other parameters that are not related to authentication configuration, which could be misleading. Consider adding an actionable message on fixing the invalid configuration
```suggestion
                    message: format!("Invalid authentication configuration. Choose either `databricks_token` or `databricks_client_id` and `databricks_client_secret`"),
```